### ISSUE TITLE
Code coverage broken when used in a logic test target

### ISSUE THREAD

**Issue by gh123man** [krzyzanowskim/CryptoSwift#739]
**Describe the bug**

On Mac OS (untested on iOS), using as a cocoapod:
Code coverage fails with the error:
```
Failed to generate coverage for target 'Pods_tests.framework' at path /Users/brian/Library/Developer/Xcode/DerivedData/cryptotest-acbtwdrzwlpbjmaszlynljvafncw/Build/Products/Debug/Pods_tests.framework/Versions/A/Pods_tests: Malformed coverage data
```

**To Reproduce**
1. Create a unit test target for your project
2. install the `CryptoSwift` pod against that target:
```

target 'tests' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!
  pod 'CryptoSwift', '1.0.0'
  # Pods for cryptotest
end
```
3. enable code coverage for "some targets" (exclude pods)
4. code coverage fails for all tests. 

**Sample code**
N/A

**Expected behavior**
Code coverage should not break when including this pod. 

**Environment (please complete the following information):**
- Swift version: 4.2
- Platform (iOS/macOS/Linux/...): MacOS 10.14.3
- Installation method (CocoaPods/Carthage/SPM/...): CocoaPods
- Project name: N/A

**Additional context**
I easily reproduced this with at totally clean test project.

**Comment by krzyzanowskim** [comment]
> Pods_tests: Malformed coverage data

I don't know. Is this CryptoSwift or Cocoapods more of the issue?

**Comment by gh123man** [comment]
Perhaps it is - however this issue only reproduces with this particular dependancy. So I am left to assume that something in this project is causing the coverage to break when running the pods target. 

I was able to reproduce this in an isolated clean project.

**Comment by gspiers** [comment]
I've had the same issue with code coverage not working in a project that includes CryptoSwift. I managed to find the problem and a work around.

I believe it relates to `SWIFT_OPTIMIZATION_LEVEL` being set to `-Owholemodule` in the xcconfig file generate by Cocoapods. There's [an OpenRadar](http://www.openradar.me/42564768) that seems to indicate Xcode code coverage won't work with `SWIFT_OPTIMIZATION_LEVEL` set to whole module.

I'm using an older version (`0.9`) so the cause was [this line](https://github.com/krzyzanowskim/CryptoSwift/blob/0.9.0/CryptoSwift.podspec#L18). Looks like this was fixed in #747 which made it into release `1.1.0`.

I'm using another dependency that is locked to CryptoSwift `0.9` so I worked around this by adding this to my Podfile:

```
post_install do |installer|
   installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
          if target.name == 'CryptoSwift' && config.name =~ /Debug/
            config.build_settings['SWIFT_OPTIMIZATION_LEVEL'] = '-Onone'
          end
        end
    end
end
```
